### ISSUE TITLE
demucs installed and walk on cmdline but fail with : No module named 'demucs.api'

### ISSUE THREAD

**Issue by daxing** [facebookresearch/demucs#577]
demucs installed and walk on cmdline but fail with : No module named 'demucs.api'
what i need to  install?

**Comment by CarlGao4** [comment]
It seems that this is related with #576

**Comment by daxing** [comment]
got it, thx!

**Comment by CarlGao4** [comment]
But I'm curious about how this happens

**Comment by ALWISHARIFF** [comment]
I am also trying to follow the docs here is the cmd logs
```
Microsoft Windows [Version 10.0.19045.3803]
(c) Microsoft Corporation. All rights reserved.

C:\Windows\system32>conda create -n "demucsnew" python=3.9
Collecting package metadata (current_repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 23.7.4
  latest version: 23.11.0

Please update conda by running

    $ conda update -n base -c defaults conda

Or to minimize the number of packages updated during conda update use

     conda install conda=23.11.0



## Package Plan ##

  environment location: C:\Users\PC\anaconda3\envs\demucsnew

  added / updated specs:
    - python=3.9


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    tzdata-2023d               |       h04d1e81_0         117 KB
    ------------------------------------------------------------
                                           Total:         117 KB

The following NEW packages will be INSTALLED:

  ca-certificates    pkgs/main/win-64::ca-certificates-2023.12.12-haa95532_0
  openssl            pkgs/main/win-64::openssl-3.0.12-h2bbff1b_0
  pip                pkgs/main/win-64::pip-23.3.1-py39haa95532_0
  python             pkgs/main/win-64::python-3.9.18-h1aa4202_0
  setuptools         pkgs/main/win-64::setuptools-68.2.2-py39haa95532_0
  sqlite             pkgs/main/win-64::sqlite-3.41.2-h2bbff1b_0
  tzdata             pkgs/main/noarch::tzdata-2023d-h04d1e81_0
  vc                 pkgs/main/win-64::vc-14.2-h21ff451_1
  vs2015_runtime     pkgs/main/win-64::vs2015_runtime-14.27.29016-h5e58377_2
  wheel              pkgs/main/win-64::wheel-0.41.2-py39haa95532_0


Proceed ([y]/n)? y


Downloading and Extracting Packages

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate demucsnew
#
# To deactivate an active environment, use
#
#     $ conda deactivate


C:\Windows\system32>conda activate demucsnew

(demucsnew) C:\Windows\system32>conda install -c conda-forge ffmpeg
Collecting package metadata (current_repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 23.7.4
  latest version: 23.11.0

Please update conda by running

    $ conda update -n base -c defaults conda

Or to minimize the number of packages updated during conda update use

     conda install conda=23.11.0



## Package Plan ##

  environment location: C:\Users\PC\anaconda3\envs\demucsnew

  added / updated specs:
    - ffmpeg


The following NEW packages will be INSTALLED:

  ffmpeg             conda-forge/win-64::ffmpeg-4.3.1-ha925a31_0

The following packages will be SUPERSEDED by a higher-priority channel:

  ca-certificates    pkgs/main::ca-certificates-2023.12.12~ --> conda-forge::ca-certificates-2023.11.17-h56e8100_0


Proceed ([y]/n)? y


Downloading and Extracting Packages

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

(demucsnew) C:\Windows\system32>python.exe -m pip install -U demucs SoundFile
Collecting demucs
  Using cached demucs-4.0.1-py3-none-any.whl
Collecting SoundFile
  Using cached soundfile-0.12.1-py2.py3-none-win_amd64.whl (1.0 MB)
Collecting dora-search (from demucs)
  Using cached dora_search-0.1.12-py3-none-any.whl
Collecting einops (from demucs)
  Using cached einops-0.7.0-py3-none-any.whl.metadata (13 kB)
Collecting julius>=0.2.3 (from demucs)
  Using cached julius-0.2.7-py3-none-any.whl
Collecting lameenc>=1.2 (from demucs)
  Using cached lameenc-1.7.0-cp39-cp39-win_amd64.whl.metadata (825 bytes)
Collecting openunmix (from demucs)
  Using cached openunmix-1.2.1-py3-none-any.whl (46 kB)
Collecting pyyaml (from demucs)
  Using cached PyYAML-6.0.1-cp39-cp39-win_amd64.whl.metadata (2.1 kB)
Collecting torch>=1.8.1 (from demucs)
  Using cached torch-2.1.2-cp39-cp39-win_amd64.whl.metadata (26 kB)
Collecting torchaudio>=0.8 (from demucs)
  Using cached torchaudio-2.1.2-cp39-cp39-win_amd64.whl.metadata (6.4 kB)
Collecting tqdm (from demucs)
  Using cached tqdm-4.66.1-py3-none-any.whl.metadata (57 kB)
Collecting cffi>=1.0 (from SoundFile)
  Using cached cffi-1.16.0-cp39-cp39-win_amd64.whl.metadata (1.5 kB)
Collecting pycparser (from cffi>=1.0->SoundFile)
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Collecting filelock (from torch>=1.8.1->demucs)
  Using cached filelock-3.13.1-py3-none-any.whl.metadata (2.8 kB)
Collecting typing-extensions (from torch>=1.8.1->demucs)
  Using cached typing_extensions-4.9.0-py3-none-any.whl.metadata (3.0 kB)
Collecting sympy (from torch>=1.8.1->demucs)
  Using cached sympy-1.12-py3-none-any.whl (5.7 MB)
Collecting networkx (from torch>=1.8.1->demucs)
  Using cached networkx-3.2.1-py3-none-any.whl.metadata (5.2 kB)
Collecting jinja2 (from torch>=1.8.1->demucs)
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting fsspec (from torch>=1.8.1->demucs)
  Using cached fsspec-2023.12.2-py3-none-any.whl.metadata (6.8 kB)
Collecting omegaconf (from dora-search->demucs)
  Using cached omegaconf-2.3.0-py3-none-any.whl (79 kB)
Collecting retrying (from dora-search->demucs)
  Using cached retrying-1.3.4-py3-none-any.whl (11 kB)
Collecting submitit (from dora-search->demucs)
  Using cached submitit-1.5.1-py3-none-any.whl.metadata (8.0 kB)
Collecting treetable (from dora-search->demucs)
  Using cached treetable-0.2.5-py3-none-any.whl
Collecting numpy (from openunmix->demucs)
  Downloading numpy-1.26.3-cp39-cp39-win_amd64.whl.metadata (61 kB)
     ---------------------------------------- 61.2/61.2 kB 34.7 kB/s eta 0:00:00
Collecting colorama (from tqdm->demucs)
  Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting MarkupSafe>=2.0 (from jinja2->torch>=1.8.1->demucs)
  Using cached MarkupSafe-2.1.3-cp39-cp39-win_amd64.whl.metadata (3.1 kB)
Collecting antlr4-python3-runtime==4.9.* (from omegaconf->dora-search->demucs)
  Using cached antlr4_python3_runtime-4.9.3-py3-none-any.whl
Collecting six>=1.7.0 (from retrying->dora-search->demucs)
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting cloudpickle>=1.2.1 (from submitit->dora-search->demucs)
  Using cached cloudpickle-3.0.0-py3-none-any.whl.metadata (7.0 kB)
Collecting mpmath>=0.19 (from sympy->torch>=1.8.1->demucs)
  Using cached mpmath-1.3.0-py3-none-any.whl (536 kB)
Using cached cffi-1.16.0-cp39-cp39-win_amd64.whl (181 kB)
Using cached lameenc-1.7.0-cp39-cp39-win_amd64.whl (148 kB)
Using cached torch-2.1.2-cp39-cp39-win_amd64.whl (192.2 MB)
Using cached torchaudio-2.1.2-cp39-cp39-win_amd64.whl (2.3 MB)
Using cached einops-0.7.0-py3-none-any.whl (44 kB)
Using cached PyYAML-6.0.1-cp39-cp39-win_amd64.whl (152 kB)
Using cached tqdm-4.66.1-py3-none-any.whl (78 kB)
Using cached filelock-3.13.1-py3-none-any.whl (11 kB)
Using cached fsspec-2023.12.2-py3-none-any.whl (168 kB)
Using cached networkx-3.2.1-py3-none-any.whl (1.6 MB)
Downloading numpy-1.26.3-cp39-cp39-win_amd64.whl (15.8 MB)
   ---------------------------------------- 15.8/15.8 MB 52.4 kB/s eta 0:00:00
Using cached submitit-1.5.1-py3-none-any.whl (74 kB)
Using cached typing_extensions-4.9.0-py3-none-any.whl (32 kB)
Using cached cloudpickle-3.0.0-py3-none-any.whl (20 kB)
Using cached MarkupSafe-2.1.3-cp39-cp39-win_amd64.whl (17 kB)
Installing collected packages: mpmath, lameenc, antlr4-python3-runtime, typing-extensions, treetable, sympy, six, pyyaml, pycparser, numpy, networkx, MarkupSafe, fsspec, filelock, einops, colorama, cloudpickle, tqdm, submitit, retrying, omegaconf, jinja2, cffi, torch, SoundFile, torchaudio, julius, dora-search, openunmix, demucs
Successfully installed MarkupSafe-2.1.3 SoundFile-0.12.1 antlr4-python3-runtime-4.9.3 cffi-1.16.0 cloudpickle-3.0.0 colorama-0.4.6 demucs-4.0.1 dora-search-0.1.12 einops-0.7.0 filelock-3.13.1 fsspec-2023.12.2 jinja2-3.1.2 julius-0.2.7 lameenc-1.7.0 mpmath-1.3.0 networkx-3.2.1 numpy-1.26.3 omegaconf-2.3.0 openunmix-1.2.1 pycparser-2.21 pyyaml-6.0.1 retrying-1.3.4 six-1.16.0 submitit-1.5.1 sympy-1.12 torch-2.1.2 torchaudio-2.1.2 tqdm-4.66.1 treetable-0.2.5 typing-extensions-4.9.0

(demucsnew) C:\Windows\system32>cd C:\Users\PC\Desktop\ai-playground\demucs

(demucsnew) C:\Users\PC\Desktop\ai-playground\demucs>code .

(demucsnew) C:\Users\PC\Desktop\ai-playground\demucs>pip show demucs
Name: demucs
Version: 4.0.1
Summary: Music source separation in the waveform domain.
Home-page: https://github.com/facebookresearch/demucs
Author: Alexandre Défossez
Author-email: defossez@fb.com
License: MIT License
Location: c:\users\pc\anaconda3\envs\demucsnew\lib\site-packages
Requires: dora-search, einops, julius, lameenc, openunmix, pyyaml, torch, torchaudio, tqdm
Required-by:

(demucsnew) C:\Users\PC\Desktop\ai-playground\demucs>ls c:\users\pc\anaconda3\envs\demucsnew\lib\site-packages
'ls' is not recognized as an internal or external command,
operable program or batch file.

(demucsnew) C:\Users\PC\Desktop\ai-playground\demucs>dir c:\users\pc\anaconda3\envs\demucsnew\lib\site-packages
 Volume in drive C has no label.
 Volume Serial Number is AA8B-3525

 Directory of c:\users\pc\anaconda3\envs\demucsnew\lib\site-packages

01/09/2024  06:13 PM    <DIR>          .
01/09/2024  06:13 PM    <DIR>          ..
01/09/2024  06:12 PM    <DIR>          antlr4
01/09/2024  06:12 PM    <DIR>          antlr4_python3_runtime-4.9.3.dist-info
01/09/2024  06:13 PM    <DIR>          cffi
01/09/2024  06:13 PM    <DIR>          cffi-1.16.0.dist-info
01/09/2024  06:13 PM    <DIR>          cloudpickle
01/09/2024  06:13 PM    <DIR>          cloudpickle-3.0.0.dist-info
01/09/2024  06:13 PM    <DIR>          colorama
01/09/2024  06:13 PM    <DIR>          colorama-0.4.6.dist-info
01/09/2024  06:13 PM    <DIR>          demucs
01/09/2024  06:13 PM    <DIR>          demucs-4.0.1.dist-info
12/11/2023  11:47 PM               152 distutils-precedence.pth
01/09/2024  06:13 PM    <DIR>          dora
01/09/2024  06:13 PM    <DIR>          dora_search-0.1.12.dist-info
01/09/2024  06:13 PM    <DIR>          einops
01/09/2024  06:13 PM    <DIR>          einops-0.7.0.dist-info
01/09/2024  06:13 PM    <DIR>          filelock
01/09/2024  06:13 PM    <DIR>          filelock-3.13.1.dist-info
01/09/2024  06:13 PM    <DIR>          fsspec
01/09/2024  06:13 PM    <DIR>          fsspec-2023.12.2.dist-info
01/09/2024  06:13 PM    <DIR>          functorch
01/09/2024  06:12 PM            11,207 isympy.py
01/09/2024  06:13 PM    <DIR>          jinja2
01/09/2024  06:13 PM    <DIR>          Jinja2-3.1.2.dist-info
01/09/2024  06:13 PM    <DIR>          julius
01/09/2024  06:13 PM    <DIR>          julius-0.2.7.dist-info
01/09/2024  06:12 PM    <DIR>          lameenc-1.7.0.dist-info
01/09/2024  06:12 PM           295,936 lameenc.cp39-win_amd64.pyd
01/09/2024  06:13 PM    <DIR>          markupsafe
01/09/2024  06:13 PM    <DIR>          MarkupSafe-2.1.3.dist-info
01/09/2024  06:12 PM    <DIR>          mpmath
01/09/2024  06:12 PM    <DIR>          mpmath-1.3.0.dist-info
01/09/2024  06:13 PM    <DIR>          networkx
01/09/2024  06:13 PM    <DIR>          networkx-3.2.1.dist-info
01/09/2024  06:13 PM    <DIR>          numpy
01/09/2024  06:12 PM                 0 numpy-1.26.3-cp39-cp39-win_amd64.whl
01/09/2024  06:13 PM    <DIR>          numpy-1.26.3.dist-info
01/09/2024  06:13 PM    <DIR>          numpy.libs
01/09/2024  06:13 PM    <DIR>          omegaconf
01/09/2024  06:13 PM    <DIR>          omegaconf-2.3.0.dist-info
01/09/2024  06:13 PM    <DIR>          openunmix
01/09/2024  06:13 PM    <DIR>          openunmix-1.2.1.dist-info
01/09/2024  06:04 PM    <DIR>          pip
01/09/2024  06:04 PM    <DIR>          pip-23.3.1-py3.9.egg-info
01/09/2024  06:04 PM    <DIR>          pkg_resources
01/09/2024  06:12 PM    <DIR>          pycparser
01/09/2024  06:12 PM    <DIR>          pycparser-2.21.dist-info
01/09/2024  06:13 PM    <DIR>          pydevd_plugins
01/09/2024  06:12 PM    <DIR>          PyYAML-6.0.1.dist-info
08/24/2023  08:59 PM               119 README.txt
01/09/2024  06:13 PM    <DIR>          retrying-1.3.4.dist-info
01/09/2024  06:13 PM            11,261 retrying.py
01/09/2024  06:04 PM    <DIR>          setuptools
01/09/2024  06:04 PM    <DIR>          setuptools-68.2.2-py3.9.egg-info
01/09/2024  06:12 PM    <DIR>          six-1.16.0.dist-info
01/09/2024  06:12 PM            34,549 six.py
01/09/2024  06:13 PM    <DIR>          soundfile-0.12.1.dist-info
01/09/2024  06:13 PM            60,756 soundfile.py
01/09/2024  06:13 PM    <DIR>          submitit
01/09/2024  06:13 PM    <DIR>          submitit-1.5.1.dist-info
01/09/2024  06:12 PM    <DIR>          sympy
01/09/2024  06:12 PM    <DIR>          sympy-1.12.dist-info
01/09/2024  06:13 PM    <DIR>          tests
01/09/2024  06:13 PM    <DIR>          torch
01/09/2024  06:13 PM    <DIR>          torch-2.1.2.dist-info
01/09/2024  06:13 PM    <DIR>          torchaudio
01/09/2024  06:13 PM    <DIR>          torchaudio-2.1.2.dist-info
01/09/2024  06:13 PM    <DIR>          torchgen
01/09/2024  06:13 PM    <DIR>          tqdm
01/09/2024  06:13 PM    <DIR>          tqdm-4.66.1.dist-info
01/09/2024  06:12 PM    <DIR>          treetable
01/09/2024  06:12 PM    <DIR>          treetable-0.2.5.dist-info
01/09/2024  06:12 PM    <DIR>          typing_extensions-4.9.0.dist-info
01/09/2024  06:12 PM           110,125 typing_extensions.py
01/09/2024  06:04 PM    <DIR>          wheel
01/09/2024  06:04 PM    <DIR>          wheel-0.41.2.dist-info
01/09/2024  06:12 PM    <DIR>          yaml
01/09/2024  06:13 PM           181,248 _cffi_backend.cp39-win_amd64.pyd
01/09/2024  06:04 PM    <DIR>          _distutils_hack
01/09/2024  06:13 PM             5,561 _soundfile.py
01/09/2024  06:13 PM    <DIR>          _soundfile_data
01/09/2024  06:12 PM    <DIR>          _yaml
01/09/2024  06:13 PM    <DIR>          __pycache__
              11 File(s)        710,914 bytes
              73 Dir(s)  142,449,541,120 bytes free

(demucsnew) C:\Users\PC\Desktop\ai-playground\demucs>dir c:\users\pc\anaconda3\envs\demucsnew\lib\site-packages\demucs
 Volume in drive C has no label.
 Volume Serial Number is AA8B-3525

 Directory of c:\users\pc\anaconda3\envs\demucsnew\lib\site-packages\demucs

01/09/2024  06:13 PM    <DIR>          .
01/09/2024  06:13 PM    <DIR>          ..
01/09/2024  06:13 PM            10,824 apply.py
01/09/2024  06:13 PM             9,650 audio.py
01/09/2024  06:13 PM             3,804 augment.py
01/09/2024  06:13 PM            17,010 demucs.py
01/09/2024  06:13 PM             3,237 distrib.py
01/09/2024  06:13 PM             1,954 ema.py
01/09/2024  06:13 PM             6,531 evaluate.py
01/09/2024  06:13 PM            30,396 hdemucs.py
01/09/2024  06:13 PM            26,577 htdemucs.py
01/09/2024  06:13 PM             3,224 pretrained.py
01/09/2024  06:13 PM                 0 py.typed
01/09/2024  06:13 PM    <DIR>          remote
01/09/2024  06:13 PM             2,811 repitch.py
01/09/2024  06:13 PM             4,727 repo.py
01/09/2024  06:13 PM             9,632 separate.py
01/09/2024  06:13 PM            16,884 solver.py
01/09/2024  06:13 PM             1,429 spec.py
01/09/2024  06:13 PM             4,835 states.py
01/09/2024  06:13 PM             3,134 svd.py
01/09/2024  06:13 PM             8,068 train.py
01/09/2024  06:13 PM            27,256 transformer.py
01/09/2024  06:13 PM             4,312 utils.py
01/09/2024  06:13 PM             9,934 wav.py
01/09/2024  06:13 PM               259 wdemucs.py
01/09/2024  06:13 PM               221 __init__.py
01/09/2024  06:13 PM               265 __main__.py
01/09/2024  06:13 PM    <DIR>          __pycache__
              25 File(s)        206,974 bytes
               4 Dir(s)  141,900,500,992 bytes free
```
**PROBLEM**
Im having this issue with demucs ModuleNotFoundError: No module named 'demucs.api' some files dont get installed with pip i dont know for what reason i am using conda i followed the docs with python 3.9 activated the environment but it fails to download the api module when installing pip install demucs
**EXPECTED BEHAVIOUR**
Pip should install all demucs files together api module file!

**Comment by CarlGao4** [comment]
The api module is introduced since 4.1.0 which hasn't been pushed to PyPI. So you need to run `python -m pip install -U --force-reinstall https://github.com/adefossez/demucs.git`. However, this version has bug currently, so you need to install from my fork: `python -m pip install -U --force-reinstall https://github.com/CarlGao4/demucs.git@4.1.0-update`

**Comment by ALWISHARIFF** [comment]
> The api module is introduced since 4.1.0 which hasn't been pushed to PyPI. So you need to run `python -m pip install -U --force-reinstall https://github.com/adefossez/demucs.git`. However, this version has bug currently, so you need to install from my fork: `python -m pip install -U --force-reinstall https://github.com/CarlGao4/demucs.git@4.1.0-update`

(demucsnew) C:\Users\PC\Desktop\ai-playground\demucs>python -m pip install -U --force-reinstall https://github.com/CarlGao4/demucs.git@4.1.0-update
Collecting https://github.com/CarlGao4/demucs.git@4.1.0-update
  ERROR: HTTP error 404 while getting https://github.com/CarlGao4/demucs.git@4.1.0-update
ERROR: Could not install requirement https://github.com/CarlGao4/demucs.git@4.1.0-update because of HTTP error 404 Client Error: Not Found for url: https://github.com/CarlGao4/demucs.git@4.1.0-update for URL https://github.com/CarlGao4/demucs.git@4.1.0

@CarlGao4 I cant install the fork you sent command fails to download install the fork whats the issue with the command?

**Comment by CarlGao4** [comment]
Sorry, I made a mistake
it should be `python -m pip install -U --force-reinstall git+https://github.com/CarlGao4/demucs.git@4.1.0-update`

**Comment by dieg1-cbegin** [comment]
Man all my files on the .local folder were deleted, now my enviroment is screwed. I don't even have numpy. Don't ever run this code
